ql/src/java/org/apache/hadoop/hive/ql/txn/compactor/Initiator.java

```diff
@@ -320,10 +320,19 @@ private boolean foundCurrentOrFailedCompactions(ShowCompactResponse compactions,
     if (compactions.getCompacts() == null) {
       return false;
     }
+
+    //In case of an aborted Dynamic partition insert, the created entry in the compaction queue does not contain
+    //a partition name even for partitioned tables. As a result it can happen that the ShowCompactResponse contains
+    //an element without partition name for partitioned tables. Therefore, it is necessary to check the partition name of
```

Comment:
I have to say I got confused here. For people like me this would be better:
> Therefore, it is necessary to check
we could have:
> Therefore, it is necessary to null check